packages/ra-core/src/routing/useGetPathForRecord.ts

```diff
@@ -51,23 +51,7 @@ export const useGetPathForRecord = <RecordType extends RaRecord = RaRecord>(
     }
     const getPathForRecord = useGetPathForRecordCallback<RecordType>(options);
 
-    // we initialize the path with the link value
-    const [path, setPath] = useState<string | false | undefined>(() => {
-        getPathForRecord({
-            record,
-            resource,
-            link,
-        }).then(resolvedLink => {
-            if (resolvedLink === false) {
-                // already set to false by default
-                return;
-            }
```

Comment:
We should probably keep this escape hatch in the effect below

Reply:
Not anymore. As we use `useQuery`, the result may change over time.